crates/rpc/rpc/src/result.rs

```diff
@@ -125,6 +126,28 @@ impl ToRpcResultExt for RethResult<Option<Block>> {
     }
 }
 
+impl ToRpcResultExt for RethResult<()> {
+    type Ok = ();
+
+    fn map_ok_or_rpc_err(self) -> RpcResult<<Self as ToRpcResultExt>::Ok> {
+        match self {
+            Ok(()) => Ok(()),
+            Err(err) => Err(internal_rpc_err(err.to_string())),
+        }
+    }
+}
+
+impl ToRpcResultExt for Result<SealedBlock, PayloadError> {
+    type Ok = SealedBlock;
+
+    fn map_ok_or_rpc_err(self) -> RpcResult<<Self as ToRpcResultExt>::Ok> {
+        match self {
+            Ok(block) => Ok(block),
+            Err(err) => Err(internal_rpc_err(err.to_string())),
+        }
+    }
+}
```

Comment:
same, we probably want

impl_to_rpc_result!(PayloadError);

here

Reply:
Done: 
https://github.com/paradigmxyz/reth/pull/4783/commits/f0247ade3d45c99c13152f34b6ed48fffa06d99b